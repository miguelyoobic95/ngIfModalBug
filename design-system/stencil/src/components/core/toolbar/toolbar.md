---
name: Toolbar
category: Core Components
---

## Attributes

|Name|Type|Default|Description|
|---|---|---|---|
|`actions`|`IActionBar[]`|   |Actions you want to set in the toolBar|
|`showActive`|`boolean`|`false`|Show active action in the toolBar|

```yoo-toolbar.html
    <yoo-toolbar class="bottom">
      <yoo-form-input attr-flex icon-suffix="yo-search"></yoo-form-input>
      <yoo-button icon="yo-plus" class="icon-only accent"></yoo-button>
    </yoo-toolbar>

    <yoo-toolbar class="accent top">
      <yoo-button class="link-transparent-accent" attr-flex text="Select All"></yoo-button>
      <yoo-button icon="yo-more" class="icon-only link-transparent-accent"></yoo-button>
    </yoo-toolbar>
```

```yoo-toolbar.css hidden
yoo-toolbar {
  margin-bottom: 40px;
}
```

## Actions attributes

```yoo-toolbar-actions.html

    <yoo-toolbar></yoo-toolbar>
```
```yoo-toolbar-actions.js

  var comp = document.querySelector('yoo-toolbar');
  comp.actions = [ { title: 'Action 1', handler: () => {}}, { title: 'Action 2', handler: () => {}}];

```

## ShowActive attributes

```yoo-toolbar-active.html

    <yoo-toolbar show-active="true"></yoo-toolbar>
```
```yoo-toolbar-active.js

  var comp = document.querySelector('yoo-toolbar');
  comp.actions = [ { title: 'Action 1', handler: () => {}}, { title: 'Action 2', handler: () => {}}];
  
```

## Interfaces 

```tsx

export interface IActionBar {
    title?: string;
    icon?: string;
    handler?: Function;
    hidden?: boolean;
}

```

## CSS

|Type|Name|Description|
|---|---|---|
|Accent|`accent`|yoo-toolbar accent|
|Top|`top`|yoo-toolbar top toolbar|
|Bottom|`bottom`|yoo-toolbar bottom toolbar|

