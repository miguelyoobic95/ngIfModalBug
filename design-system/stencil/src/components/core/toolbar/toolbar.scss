

@mixin circle-border-color($class, $color) {
    &.border-#{$class} {
        border-color: cssvar($color);
    }
}

@mixin circle-background-color($class, $color) {
    &.#{$class} {
        background: cssvar($color);
    }
}

:host {
    display: block;
    .container {
        width: 100%;
        padding: 0.75rem 1rem;
        /deep/>*:not(:last-child) {
            margin-right: 1rem;
        }

        .actions { 
            color: cssvar(light);
            width: 100%;
            height: 80px;

            .action {
                text-align: center;
                cursor: pointer;
                
                .circle-container {
                    position: relative;
                    height: 50px;
                    width: 46px;
                    margin: auto;
                
                    .circle {
                        color: cssvar(stable);
                        position: absolute;
                        top: 3px;
                        left: 3px;
                        width: 40px;
                        height: 40px;
                        border-radius: 50%;
                        line-height: 40px;
                        margin: auto;
                        margin-bottom: 5px;
                        border-width: $border-width;
                        font-size: 1rem;
                        @include circle-background-color(accent, accent);
                        @include circle-background-color(danger, danger);
                        @include circle-background-color(success, success);
                        @include circle-background-color(info, info);
                        @include circle-background-color(warning, warning);
                        @include circle-background-color(dark-60, dark-60);
                    }

                    &:hover {
                        .circle-border {
                            display: none;
                        }
                    }
                    .circle-border {
                        position: absolute;
                        border-radius: 50%;
                        border-width: 1px;
                        border-style: solid;
                        top: 0;
                        width: 46px;
                        height: 46px;
                        @include circle-border-color(accent, accent);
                        @include circle-border-color(danger, danger);
                        @include circle-border-color(success, success);
                        @include circle-border-color(info, info);
                        @include circle-border-color(warning, warning);
                        @include circle-border-color(dark-60, dark-60);
                    }
                    
                }
                .label {
                    @include text-truncate();
                    padding: 2px 10px;
                    color: cssvar(dark-60);
                    font-size: 0.8125rem;
                }

                &.active {
                    .circle-container {
                        .circle {
                            background: cssvar(success);
                            color: cssvar(light);
                        }
                    }
                }   
            }
        }
    }
}

:host(.accent) {
    .container {
        background: cssvar(accent-10);
        border-bottom: 1px solid cssvar(accent);
    }
}

:host(.top) {
    .container {
        border-bottom: 1px solid cssvar(dark-20);
    }
}

:host(.bottom) {
    .container {
        border-top: 1px solid cssvar(dark-20);
    }
}