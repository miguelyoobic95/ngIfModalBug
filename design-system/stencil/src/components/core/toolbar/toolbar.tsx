import { Component, Prop, State, Element } from '@stencil/core';
import { IActionBar } from '@shared/interfaces';

@Component({
    tag: 'yoo-toolbar',
    styleUrl: 'toolbar.scss',
    scoped: true
})
export class YooToolbarComponent {

    @Prop() actions: IActionBar[];
    @Prop() showActive: boolean = false;

    @State() activeAction: IActionBar;

    @Element() host: HTMLStencilElement;

    public colors: Array<string> = ['accent', 'danger', 'success', 'info', 'warning', 'dark-60'];

    onClick(action: IActionBar) {
        if (action && action.handler) {
            action.handler();
            if (this.showActive) {
                this.activeAction = action;
            }
        }
    }

    getColor(i: number): string {
        return this.colors[i % this.colors.length];
    }

    render() {
        return this.actions ? (
            <div class="container" attr-layout="row" attr-layout-align="space-around center">
                <div class="actions" attr-layout="row" attr-layout-align="space-between">
                    {this.actions.map((a, i) =>
                        <div onClick={this.onClick.bind(this, a)} class={'action ' + (this.activeAction === a ? 'active' : '')}>
                            <div class="circle-container">
                                <div class={'circle-border border-' + this.getColor(i)} ></div>
                                <div class={'circle ' + this.getColor(i)}><i class={a.icon}></i></div>
                            </div>
                            <div class="label">{a.title}</div>
                        </div>
                    )}
                </div>
            </div>
        ) :
            (<div class="container" attr-layout="row" attr-layout-align="space-around center">
                <slot />
            </div>);
    }
}
