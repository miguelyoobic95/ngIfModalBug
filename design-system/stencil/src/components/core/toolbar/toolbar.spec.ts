import { TestWindow } from '@stencil/core/testing';
import { YooToolbarComponent } from './toolbar';

describe('YooToolbarComponent', () => {
    it('should build', () => {
        expect(new YooToolbarComponent()).toBeTruthy();
    });

    describe('property rendering', () => {
        it ('should render', async () => {
            let window = new TestWindow();
          let element = await window.load({
                components: [YooToolbarComponent],
                html: '<yoo-toolbar></yoo-toolbar>'
            });
            expect(element).toBeTruthy();
            expect(element).toMatchSnapshot();
        });
        it ('should render any content in the slot', async () => {
            let window = new TestWindow();
          let element = await window.load({
                components: [YooToolbarComponent],
                html: '<yoo-toolbar><div>Hey</div></yoo-toolbar>'
            });
            expect(element).toMatchSnapshot();
        });
        it ('should render with actions', async () => {
            let window = new TestWindow();
          let element = await window.load({
                components: [YooToolbarComponent],
                html: '<yoo-toolbar></yoo-toolbar>'
            });
            element.actions = [ { title: 'Action 1'}, { title: 'Action 2'}];
            await window.flush();
            expect(element).toMatchSnapshot();
            expect(element.querySelectorAll('.action')).toHaveLength(2);
        });
        it ('should highlight the active action and call the handler', async () => {
            let window = new TestWindow();
            let element = await window.load({
                components: [YooToolbarComponent],
                html: '<yoo-toolbar></yoo-toolbar>'
            });
            let handler = jest.fn();
            element.actions = [{ title: 'Action 1', handler: handler}, { title: 'Action 2', handler: handler}];
            element.showActive = true;
            await window.flush();
            let activeAction = element.querySelectorAll('.action')[0];
            await activeAction.click();
            expect(handler).toHaveBeenCalled();
            await window.flush();
            expect(element.querySelectorAll('.active')).toHaveLength(1);
        });
    });
    describe('helper methods', () => {
        it('should get the color corresponding to the index', async () => {
            let element = new YooToolbarComponent();
            let color = element.getColor(0);
            expect(color).toEqual('accent');
        });
    });

});
