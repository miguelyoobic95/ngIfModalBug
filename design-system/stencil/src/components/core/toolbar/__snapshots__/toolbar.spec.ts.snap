// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooToolbarComponent property rendering should render 1`] = `
<yoo-toolbar
  class="hydrated"
  data-yoo-toolbar-host=""
>
  
  <div
    attr-layout="row"
    attr-layout-align="space-around center"
    class="container"
    data-yoo-toolbar=""
    data-yoo-toolbar-slot=""
  >
    
  </div>
</yoo-toolbar>
`;

exports[`YooToolbarComponent property rendering should render any content in the slot 1`] = `
<yoo-toolbar
  class="hydrated"
  data-yoo-toolbar-host=""
>
  
  
  <div
    attr-layout="row"
    attr-layout-align="space-around center"
    class="container"
    data-yoo-toolbar=""
    data-yoo-toolbar-slot=""
  >
    
    <div>
      Hey
    </div>
  </div>
</yoo-toolbar>
`;

exports[`YooToolbarComponent property rendering should render with actions 1`] = `
<yoo-toolbar
  class="hydrated"
  data-yoo-toolbar-host=""
>
  
  <div
    attr-layout="row"
    attr-layout-align="space-around center"
    class="container"
    data-yoo-toolbar=""
    data-yoo-toolbar-slot=""
  >
    <div
      attr-layout="row"
      attr-layout-align="space-between"
      class="actions"
      data-yoo-toolbar=""
    >
      <div
        class="action"
        data-yoo-toolbar=""
      >
        <div
          class="circle-container"
          data-yoo-toolbar=""
        >
          <div
            class="circle-border border-accent"
            data-yoo-toolbar=""
          />
          <div
            class="circle accent"
            data-yoo-toolbar=""
          >
            <i
              data-yoo-toolbar=""
            />
          </div>
        </div>
        <div
          class="label"
          data-yoo-toolbar=""
        >
          Action 1
        </div>
      </div>
      <div
        class="action"
        data-yoo-toolbar=""
      >
        <div
          class="circle-container"
          data-yoo-toolbar=""
        >
          <div
            class="circle-border border-danger"
            data-yoo-toolbar=""
          />
          <div
            class="circle danger"
            data-yoo-toolbar=""
          >
            <i
              data-yoo-toolbar=""
            />
          </div>
        </div>
        <div
          class="label"
          data-yoo-toolbar=""
        >
          Action 2
        </div>
      </div>
    </div>
  </div>
</yoo-toolbar>
`;
